app/user/rest.py

```diff
@@ -105,19 +105,15 @@ def handle_integrity_error(exc):
 
 @user_blueprint.route("", methods=["POST"])
 def create_user():
-    # import pdb; pdb.set_trace()
-    user_to_create, errors = create_user_schema.load(request.get_json())
```

Comment:
This is a good observation! And the behavior will be similar as before for reasons I can explain. Marshmallow, in the new major version, will fail (i.e. throw an exception) on the `load` method invocation ([v3 doc for `load`](https://marshmallow.readthedocs.io/en/stable/api_reference.html#marshmallow.Schema.load)) whereas previously, it would rather report the errors on load rather than raising an exception ([v2 doc for `load`](https://marshmallow.readthedocs.io/en/2.x-line/api_reference.html#marshmallow.Schema.load)) (note: there was a strict option in v2 but it seems it applied on serialization/marshalling per the doc and not on deserialization, the latter being what's happening in this code). Hence, [the new major version 3 is strict](https://marshmallow.readthedocs.io/en/stable/upgrading.html#schemas-are-always-strict) and less flexible if we can say. 

The `ValidationError` exception will carry the errors in its `messages` fields. So when the call is made to `load` here and there are errors, it will raise the exception and bubble up to the Flask error handlers layer. We have one error handler for [`ValidationError` located here](https://github.com/cds-snc/notification-api/blob/8ff1e8130d137b8448cdcf70118a1d871724cf93/app/errors.py#L55) with the following code:

```
    @blueprint.errorhandler(ValidationError)
    def marshmallow_validation_error(error):
        current_app.logger.info(error)
        return jsonify(result="error", message=error.messages), 400
```

So this will report a `400` and carry the error messages back the API user.

Previously, we were checking for the errors returned from the second tuple members, then converting it into an `InvalidRequest` and raising it, which would be caught by [this error handler instead](https://github.com/cds-snc/notification-api/blob/8ff1e8130d137b8448cdcf70118a1d871724cf93/app/errors.py#L70), but with similar results.

So overall, there is less error code management but also less transparency, one could argue (and more mental effort to understand how this is managed as demonstrated here), leading us to wonder how this is all handled. We could catch the `ValidationError` and convert it into a `InvalidRequest`, but that might be less idiomatic of the language and its libraries overall. 

What do you think? does that make sense?

Reply:
Yes that makes sense.  I thought maybe we were suppressing errors but it sounds like we are not, so no issue on my end!